xe2: gemm: Update BMG int4 row major N: 1-64 strategy

Yejing-Lai opened the pull request:
Update BMG int4 row major N: 1-64 strategy. Mainly improved the case on qwen/llama models.

Yejing-Lai commented:
@CaoZhongZ @Simonsays095

Simonsays095 commented:
make test perf-gpu
set primitive=gpu:gemm

Yejing-Lai commented:
Hi @Simonsays095. May I know how the performance model chooses the optimal strategy when one policy is a W model and the other is an E model?

I've tried adding an old kernel(W model) for N=25-32 or switching it to an E model, but the performance model always chooses the old kernels, even when the new kernel performs better.

Simonsays095 commented:
@Yejing-Lai It's not intended to mix performance models. The way the different models work, mixing W and E will almost always select W, since the scores are much lower. I recommend changing the old kernel to model E using `ktool ... --model E --refit <kernel id>`

Yejing-Lai commented:
Hi @Simonsays095. I updated the kernel for N=33-48, and added a new kernel for N=25-32 with M<<K case, which resolves some of the regression. Could you trigger the CI again?

Yejing-Lai commented:
Hi @Simonsays095. Thanks for your recommendation. I  have tried refitting the old kernel but it did not fix the issue. I choose option 1( copy it to the end of the catalog and change 'G' to 'H').
